src/main/java/org/isf/utils/ExceptionUtils.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Open Hospital (www.open-hospital.org)
+ * Copyright © 2006-2021 Informatici Senza Frontiere (info@informaticisenzafrontiere.org)
+ *
+ * Open Hospital is a free and open source software for healthcare data management.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * https://www.gnu.org/licenses/gpl-3.0-standalone.html
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+package org.isf.utils;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+public class ExceptionUtils {
+
+	public static String retrieveExceptionStacktrace(Exception e) {
+		StringWriter writer = new StringWriter();
+		PrintWriter printWriter = new PrintWriter(writer);
+		e.printStackTrace(printWriter);
+		printWriter.flush();
+		return writer.toString();
+	}
```

Comment:
I don't see the purpose/need for this class.   slf4j logging methods will dump the stack trace automatically if the last parameter is an exception.   So code like this:
<pre>
LOGGER.error(e.getMessage());
LOGGER.error(ExceptionUtils.retrieveExceptionStacktrace(e));
</pre>

can be replaced by:
<pre>
LOGGER.error(e.getMessage(), e);
</pre>

If you want just the stacktrace then use:
<pre>
LOGGER.error("", e);
</pre>

There are 3 occurrences where this method is used:

- orig.isf.telemetry.daemon.TelemetryDaemon  (2 occurrences)
- org.isf.telemetry.envdatacollector.DataColletorProviderService